doc/manual/rl-next/pipe-operators.md

```diff
@@ -0,0 +1,14 @@
+---
+synopsis: Add `pipe-operators` experimental feature
+prs:
+- 11131
+---
+
+This is a draft implementation of [RFC 0148](https://github.com/NixOS/rfcs/pull/148).
+
+The `pipe-operators` experimental feature adds `<|` and `|>` operators to the Nix language.
+*a* `|>` *b* is equivalent to the function application *b* *a*, and
```

Comment:
thought: "a into b", should we give it a name like that? Could be nice.

Reply:
:shrug: The other direction is a little harder to name, I think? I don't know if this is important.

Reply:
_"into"_ follows nicely from the more explicit "_a_ piped into _b_", though I'm not sure about the other direction (`<|`).

Maybe that could be read as _"from"_, as in `a <| b` is "_a_ piped from _b_"; although I'd personally just read it backwards as "_b_ piped into _a_".

Reply:
"taking"?

Reply:
Or ‘of’?

This seems like tutorial stuff, not reference stuff; these names would be entirely non-normative and not necessary for understanding the language from first principles. So does it belong in the reference documentation, or in release notes, or both or neither?

Reply:
> Or ‘of’?
> 
> This seems like tutorial stuff, not reference stuff; [...] entirely non-normative

Certainly. It was just a

> thought

I figured it'd be good to talk about this.

------

I think I like these two, because they're short and literal, with a word per symbol, in order.

- `|>`  pipe into
- `<|` back pipe

Reply:
Late to the party, but FWIW in the current Lix draft I call them pipe left / pipe right in the code and tend towards calling them pipe (forward) and pipe backward in the documentation. (The forward being implicit and sometimes omitted)

Reply:
Just to add some more ideas into the pot, a plumber would usually talk about pipes "feeding" or "supplying" and then "draining" into the wastewater pipes.

E.g:
- "this tap is fed from the cold water supply"
- "the sink drains down a 50mm pipe"
- "the bath is supplied hot water from pipe A"
- "pipe B supplies the bath with cold water"
- "pipe C feeds the shower with cold water"
- "the shower is fed hot water by pipe D"

I don't think this will sound natural in the context of programming, but I thought it was worth reminiscing on my background in construction 😅

I agree "Pipe left/backward" and "pipe right/forward/into" should be fairly natural, simple and obvious